docs/observability.md

```diff
@@ -0,0 +1,84 @@
+# Observability with Grafana Dashboard
+
+This document explains how to use the `grafana-dashboard` make target to install and configure observability tools, including Prometheus and Grafana, on your Kubernetes cluster. The setup uses Helm charts to install Prometheus and Grafana, provides a Prometheus data source, and applies a Grafana dashboard configuration.
+
+## Prerequisites
+
+Ensure the following tools are installed on your local machine:
+- **Kubernetes**: A running Kubernetes cluster.
+- **kubectl**: To manage the cluster.
+- **Helm**: To install and manage Helm charts for Prometheus and Grafana.
+
+You should also have access to the Kubernetes cluster's kubeconfig file (`test-cluster-kubeconfig.yaml`), which will be used for running the make target.
+
+## Steps to Install the Grafana Dashboard
+
+### 1. Build and Set Up the Cluster (Optional)
+If you haven’t already set up your Kubernetes cluster with the necessary CSI driver and Prometheus metrics services, you can do so by running the following command:
+```bash
+make mgmt-and-capl-cluster
+```
+This command creates a management cluster and CAPL (Cluster API for Linode) cluster, installs the Linode CSI driver, and applies the necessary configurations to expose the CSI metrics.
+
+### 2. Run the Grafana Dashboard Setup
+The make target `grafana-dashboard` will install Prometheus and Grafana in a Kubernetes namespace (`monitoring` by default) and configure a Grafana dashboard from a local JSON file. To execute this setup, run:
+
+```bash
+make grafana-dashboard
+```
```

Comment:
Could we add some configuration options here that can be passed into the make target such as username, password, and data-retention period for prometheus?

Reply:
I've added this and updated the makefile and the install script